src/pe/utils.rs

```diff
@@ -74,3 +76,24 @@ pub fn try_name<'a>(bytes: &'a [u8], rva: usize, sections: &[section_table::Sect
         }
     }
 }
+
+pub fn get_data<'a, T>(bytes: &'a [u8], sections: &[section_table::SectionTable], directory: &DataDirectory, file_alignment: u32) -> Result<T, DataDirectoryConversionError>
+    where T: scroll::ctx::TryFromCtx<'a, scroll::Endian, Size = usize, Error = scroll::Error>  {
+    let rva = directory.virtual_address as usize;
+    let offset = find_offset(rva, sections, file_alignment)
+        .ok_or(DataDirectoryConversionError::MissingOffset(directory.virtual_address))?;
+    let result: T = bytes.pread_with(offset, scroll::LE).map_err(DataDirectoryConversionError::ReadError)?;
+    Ok(result)
+}
+
+#[derive(Debug)]
+pub enum DataDirectoryConversionError {
```

Comment:
No more error enums please. It just makes working with code more annoying. It also doesn’t implement Error, etc. please use the goblin error; if you can’t reuse one of the errors without losing semantics, just use the Custom string option.